bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: Block Kafka Creation, Service account creation during the migration phase & switched

## Description
https://issues.redhat.com/browse/MGDSTRM-8484

Block Kafka creation and service account creation during the migration - apart from a selected org (this org would be used for testing during the migration)

Through the use of access-list-configuration.yaml we may allow access to specified organisations for kafka, and service account creations.

## Verification Steps
__Step 1. Check If the Flags are available:__
- `make binary`
- ./kas-fleet-manager serve -h
- Ensure `--enable-access-list `, and `--access-list-config-file` are listed.

__Step 2. Check the configurations are functional:__
- Start a [Cluster](https://qaprodauth.cloud.redhat.com/)
- run `ngrok http 8000`
- Find your cluster `ocm list clusters --parameter search="name like '<ClusterName>'”`
- `ocm get /api/clusters_mgmt/v1/clusters/<ClusterID> | jq ' .name, .id, .multi_az, .cloud_provider.id, .provider_type '`
- run `./kas-fleet-manager serve --enable-access-list --public-host-url=<ngrok address>`
- run commands such as: `curl -v -XPOST -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/kafkas_mgmt/v1/kafkas?async=true -d '{ "region": "us-east-1", "cloud_provider": "aws",  "name": "serviceapi", "multi_az":true}'` to create artifacts.
- these commands should function correctly if your orgId is added to access-list-configuration.yaml

__Step 3. :__
- remove your orgId from access-list-configuration.yaml
- run `./kas-fleet-manager serve --enable-access-list --public-host-url=<ngrok address>`
- run commands such as: `curl -v -XPOST -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/kafkas_mgmt/v1/kafkas?async=true -d '{ "region": "us-east-1", "cloud_provider": "aws",  "name": "serviceapi", "multi_az":true}'` to create artifacts.
- this time you will be unable to create artifacts and will be presented with a message similar to:

```
{"id":"18","kind":"Error","href":"/api/kafkas_mgmt/v1/errors/18","code":"KAFKAS-MGMT-18","reason":"organisation '13640203' is not authorized to access the service during the current service maintanence.","operation_id":"cagarif13hooqiieoomg"}

```

## Checklist (Definition of Done)
